snuba/datasets/querylog.py

```diff
@@ -0,0 +1,94 @@
+from datetime import timedelta
+from snuba import settings
+from snuba.clickhouse.columns import (
+    Array,
+    ColumnSet,
+    DateTime,
+    Enum,
+    Float,
+    LowCardinality,
+    Mapping,
+    Nested,
+    Nullable,
+    String,
+    UInt,
+    UUID,
+)
+
+from snuba.datasets.dataset import TimeSeriesDataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.schemas.tables import MergeTreeSchema
+from snuba.datasets.table_storage import TableWriter, KafkaStreamLoader
+from snuba.datasets.querylog_processor import QuerylogProcessor
+from snuba.query.extensions import QueryExtension
+from snuba.query.project_extension import ProjectExtension, ProjectExtensionProcessor
+from snuba.query.timeseries import TimeSeriesExtension
+
+
+class QuerylogDataset(TimeSeriesDataset):
+    def __init__(self) -> None:
+        status_type = Enum([("success", 0), ("error", 1), ("rate-limited", 2)])
+
+        columns = ColumnSet(
+            [
+                ("request", String()),
+                ("referrer", LowCardinality(String())),
+                ("dataset", LowCardinality(String())),
+                ("projects", Array(UInt(64))),
+                ("organization", Nullable(UInt(64))),
+                ("timestamp", DateTime()),
+                ("duration_ms", UInt(32)),
+                ("status", status_type),
+                (
+                    "clickhouse_queries",
+                    Nested(
+                        [
+                            ("sql", String()),
+                            ("status", status_type),
+                            ("final", UInt(8)),
+                            ("cache_hit", UInt(8)),
+                            ("sample", Float(32)),
+                            ("max_threads", UInt(8)),
+                            ("duration_ms", UInt(32)),
+                            ("trace_id", Nullable(UUID())),
+                        ]
+                    ),
+                ),
+            ]
+        )
+
+        schema = MergeTreeSchema(
+            columns=columns,
+            local_table_name="querylog_local",
+            dist_table_name="querylog_dist",
+            order_by="(timestamp)",
+            partition_by="(toMonday(timestamp))",
```

Comment:
please add a sample clause, just in case, because we cannot add it later.

Reply:
I would be hesitant to add a sampling clause to this. It's an internal dataset. If we require sampling this data to get anything remotely accurate out of it, I believe we'd be doing something incorrect.